### PR TITLE
Vermont Medical Exemption

### LINKED ISSUE #2624: Refine Vermont agi subtraction
Add the following five types of subtractions to `vt_agi_subtractions`

**Interest Income from U.S. Obligations (us_govt_interest)**
Legal Code: https://legislature.vermont.gov/statutes/section/32/151/05811 Titl. 32 V.S.A. § 5811(21)(B)(i)
_Tax Form:_ 
https://tax.vermont.gov/sites/tax/files/documents/IN-111-2022.pdf  2022 Form IN-111 Vermont Income Tax Return Line 2
https://tax.vermont.gov/sites/tax/files/documents/IN-112-2022.pdf 2022 Schedule IN-112 Vermont Tax Adjustments and Credits Part 1
_Instruction:_ 
https://tax.vermont.gov/sites/tax/files/documents/IN-112%20Instr-2022.pdf  Instruction for 2022 SCHEDULE IN-112

**Medical Expenses Deduction (medical_expense)**
Legal Code: https://legislature.vermont.gov/statutes/section/32/151/05811 Titl. 32 V.S.A. § 5811(21)(C)(iv)
_Tax Form:_ 
https://tax.vermont.gov/sites/tax/files/documents/IN-111-2022.pdf  2022 Form IN-111 Vermont Income Tax Return Line 2
https://tax.vermont.gov/sites/tax/files/documents/IN-112-2022.pdf 2022 Schedule IN-112 Vermont Tax Adjustments and Credits Part 1
_Instruction:_ 
https://tax.vermont.gov/sites/tax/files/documents/IN-112%20Instr-2022.pdf  Instruction for 2022 SCHEDULE IN-112

**Student Loan Interest (student_loan_interest)**
Legal Code: https://legislature.vermont.gov/statutes/section/32/151/05811 Titl. 32 V.S.A. § 5811(21)(B)(vi)
_Tax Form:_ 
https://tax.vermont.gov/sites/tax/files/documents/IN-111-2022.pdf  2022 Form IN-111 Vermont Income Tax Return Line 2
https://tax.vermont.gov/sites/tax/files/documents/IN-112-2022.pdf 2022 Schedule IN-112 Vermont Tax Adjustments and Credits Part 1
_Instruction:_ 
https://tax.vermont.gov/sites/tax/files/documents/IN-112%20Instr-2022.pdf  Instruction for 2022 SCHEDULE IN-112

**Capital Gains Exclusion** `capital_gains_excluded_from_taxable_income`
Legal Code: https://legislature.vermont.gov/statutes/section/32/151/05811 Titl. 32 V.S.A. § 5811(21)(B)(ii)
Tax Form:
https://tax.vermont.gov/sites/tax/files/documents/IN-111-2022.pdf 2022 Form IN-111 Vermont Income Tax Return Line 2
https://tax.vermont.gov/sites/tax/files/documents/IN-112-2022.pdf 2022 Schedule IN-112 Vermont Tax Adjustments and Credits Part 1
Instruction:
https://tax.vermont.gov/sites/tax/files/documents/IN-112%20Instr-2022.pdf Instruction for 2022 SCHEDULE IN-112

**Retirement Benefits Exempt from Vermont Taxation**
Legal Code: https://legislature.vermont.gov/statutes/section/32/151/05811 Titl. 32 V.S.A. § 5811(21)(B)(iv)
Tax Form:
https://tax.vermont.gov/sites/tax/files/documents/IN-111-2022.pdf 2022 Form IN-111 Vermont Income Tax Return Line 2
https://tax.vermont.gov/sites/tax/files/documents/IN-112-2022.pdf 2022 Schedule IN-112 Vermont Tax Adjustments and Credits Part 1
Instruction:
https://tax.vermont.gov/sites/tax/files/documents/IN-112%20Instr-2022.pdf Instruction for 2022 SCHEDULE IN-112

### PR DESCRIPTION
**Pull request by mzx-4real**
Related to #2624 

### Summary
🆕📄🐛

This pull request adds a new feature to the policy engine: the Vermont medical expense deduction, which reduces the taxable income for Vermont residents who have high medical expenses. It also adds a new parameter file and two unit test files for the Vermont subtractions from adjusted gross income, which include the medical expense deduction and other sources of income that are excluded from taxation in Vermont. Additionally, it renames a test input variable, deletes an unused variable file, and removes a reference to an unimplemented variable.

> _Sing, O Muse, of the skillful coder who devised_
> _A new feature for the policy engine, the Vermont deduction_
> _That reduces the tax burden of the mortals who incur_
> _Large medical expenses, exceeding their exemptions._

### Walkthrough
*  Add a new feature, the Vermont Medical Expense Deduction, to the policy engine ([link](https://github.com/PolicyEngine/policyengine-us/pull/2996/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R4))
*  Create a new parameter file for the Vermont subtractions from adjusted gross income, which include the medical expense deduction and other sources of income that are excluded from taxation in Vermont ([link](https://github.com/PolicyEngine/policyengine-us/pull/2996/files?diff=unified&w=0#diff-8299b4ee7331e5c8c2dfaf432847d46016da57b0dc4fc86cb0cf63b94c52bd46R1-R20))
*  Create a new variable file for the Vermont medical expense deduction, which calculates the deduction amount as the excess of the federal deduction over the sum of the standard deduction and the personal exemptions, or zero if the excess is negative ([link](https://github.com/PolicyEngine/policyengine-us/pull/2996/files?diff=unified&w=0#diff-672e0924f829ef486e0b8e95360ca65eadacb1999a477294f4a88d36d132306eR1-R31))
*  Create a new variable file for the Vermont subtractions from adjusted gross income, which uses the `adds` attribute to specify the parameter list that contains the sources of income that are subtracted from the adjusted gross income in Vermont ([link](https://github.com/PolicyEngine/policyengine-us/pull/2996/files?diff=unified&w=0#diff-e53b6fc7915e4489ef8722758b6e370a813ea98e387583dc2b78d97767d33a91R1-R18))
*  Rename the `subtracts` attribute in the `vt_agi` variable file, to match the new parameter name `vt_subtractions` ([link](https://github.com/PolicyEngine/policyengine-us/pull/2996/files?diff=unified&w=0#diff-c174b78e440841750174c9f041ab5510a7b82fe4158a8c204e567d494ce3bb78L14-R14))
*  Delete the unused `vt_agi_subtractions` variable file ([link](https://github.com/PolicyEngine/policyengine-us/pull/2996/files?diff=unified&w=0#diff-3d3e0e0508b8f3ca30470933ed023041247e2764151f8a7753d06b4ff9d4a842))
*  Remove the `vt_income_tax` variable from the list of state income taxes in the `state_income_tax` variable file, as a temporary fix until the variable is implemented ([link](https://github.com/PolicyEngine/policyengine-us/pull/2996/files?diff=unified&w=0#diff-af0d2b84ba0adacb1b6e826175576659344b80a34710502f7791da61cbd9c600L38))
*  Add unit tests for the Vermont medical expense deduction and the Vermont subtractions from adjusted gross income, using different scenarios and inputs, and testing the state definition by using a different state code ([link](https://github.com/PolicyEngine/policyengine-us/pull/2996/files?diff=unified&w=0#diff-5ead6796cda764ecd83d02353656109ce22b9336fa0c3c0d5bfac020754c99cdR1-R39), [link](https://github.com/PolicyEngine/policyengine-us/pull/2996/files?diff=unified&w=0#diff-c82d3398f11eca599d774b9ff9de4a75a889ba21bdc671e90859e4a408d3ce9aR1-R47))
*  Rename the input variable `vt_agi_subtractions` to `vt_subtractions` in the unit test file for the Vermont adjusted gross income, to match the new parameter name ([link](https://github.com/PolicyEngine/policyengine-us/pull/2996/files?diff=unified&w=0#diff-aa126afd3fd6d854e97157e4ca225c16c3a15dcee782e8b3b26957a4141419f8L18-R18), [link](https://github.com/PolicyEngine/policyengine-us/pull/2996/files?diff=unified&w=0#diff-aa126afd3fd6d854e97157e4ca225c16c3a15dcee782e8b3b26957a4141419f8L30-R30))

### CODE REVIEW

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+      - Vermont Medical Expense Deduction.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
      - Vermont medical expense deduction.
```

File: policyengine_us/variables/gov/states/vt/tax/income/adjusted_gross_income/subtractions/vt_subtractions.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+class vt_subtractions(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Vermont subtractions"
+    unit = USD
+    documentation = "Subtractions from Vermont adjusted gross income"
+    definition_period = YEAR
+    defined_for = StateCode.VT
+    reference = (
+        "https://tax.vermont.gov/sites/tax/files/documents/IN-112-2022.pdf#page=1"  # PART 1 SUBTRACTIONS TO FEDERAL ADJUSTED GROSS INCOME
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/vt/tax/income/adjusted_gross_income/subtractions/vt_subtractions.py:13]:
```suggestion
        "https://tax.vermont.gov/sites/tax/files/documents/IN-112-2022.pdf#page=1",  # PART 1 SUBTRACTIONS TO FEDERAL ADJUSTED GROSS INCOME
```

File: policyengine_us/variables/gov/states/vt/tax/income/adjusted_gross_income/subtractions/vt_subtractions.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        "https://legislature.vermont.gov/statutes/section/32/151/05811"  # Titl. 32 V.S.A. § 5811(21)(B)(i), (C)(iv), (B)(vi), (B)(ii), (B)(iv)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/vt/tax/income/adjusted_gross_income/subtractions/vt_subtractions.py:14]:
```suggestion
        "https://legislature.vermont.gov/statutes/section/32/151/05811"m  # Titl. 32 V.S.A. § 5811(21)(B)(i), (C)(iv), (B)(vi), (B)(ii), (B)(iv)
```

File: policyengine_us/variables/gov/states/vt/tax/income/adjusted_gross_income/subtractions/vt_subtractions.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/vt/tax/income/adjusted_gross_income/subtractions/vt_subtractions.py:1]: file broken

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/variables/gov/states/vt/tax/income/adjusted_gross_income/subtractions/vt_medical_expense_deduction.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    reference = (
+        "https://legislature.vermont.gov/statutes/section/32/151/05811"  # Titl. 32 V.S.A. § 5811(21)(C)(iv)
+        "https://tax.vermont.gov/sites/tax/files/documents/IN-112%20Instr-2022.pdf#page=2",  # Instruction for 2022 SCHEDULE IN-112 - MEDICAL DEDUCTION WORKSHEET
+    )
+    unit = USD
+    defined_for = StateCode.VT
+    documentation = "Vermont medical expenses deducted from taxable income."
+
+    def formula(tax_unit, period, parameters):
+        # Get federal medical expense deduction (Worksheet line 1a).
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/vt/tax/income/adjusted_gross_income/subtractions/vt_medical_expense_deduction.py:18]:
```suggestion
        # Get federal medical expense deduction (Worksheet line 1a).
        # This points to federal Form 1040, Schedule A, Line 4, which is the deduction itself
        # (not the expenses).
        # The law says "an amount equal to the itemized deduction for medical expenses
        # taken at the federal level by the taxpayer" but does not state whether the filer
        # must have itemized on the federal return to claim it.
        # We assume they do not have to, based on the form.
```

File: policyengine_us/parameters/gov/states/vt/tax/income/agi/subtractions.yaml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+description: Vermont counts these sources as subtractions.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/vt/tax/income/agi/subtractions.yaml:1]:
```suggestion
description: Vermont subtracts these sources from adjusted gross income.
```

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**